codes/c/chapter_hashing/hash_map_chaining.c

```diff
@@ -0,0 +1,235 @@
+/**
+ * File: hash_map_chaining.c
+ * Created Time: 2023-08-29
+ * Author: SenMing (1206575349@qq.com)
+ */
+
+#include "../utils/common.h"
+
+/* 键值对 int->string */
+typedef struct node {
+    int key;
+    char* val;
+    struct node* next;              // 下一个元素
+} Node;
+
+typedef struct pair {
+    Node* node;                     // 键值对节点
+} Pair;
+
+/* 基于数组简易实现的哈希表 */
+typedef struct ArrayHashMap  {
+    Pair* buckets;                  // 桶数组
+    int size;                       // 键值对数量
+    int capacity;                   // 哈希表容量
+    int extendRatio;                // 扩容倍数
+    double loadThres;               // 触发扩容的负载因子阈值
+} ArrayHashMap;
+
+// 函数声明
+void extend(ArrayHashMap* hashmap);
+
+/* 初始化桶数组 */
+ArrayHashMap* InitArrayHashMap(int tableSize) {
+    // 为哈希表分配空间
+    ArrayHashMap* hashmap = (ArrayHashMap*)malloc(sizeof(ArrayHashMap));
+    
+    // 初始化数组
+    hashmap->buckets = (Pair*)malloc(sizeof(Pair) * tableSize);
+    memset(hashmap->buckets, 0, sizeof(Pair) * tableSize);
+
+    hashmap->capacity = tableSize;
+    hashmap->size = 0;
+    hashmap->extendRatio = 2;
+    hashmap->loadThres = 2.0 / 3;
+
+    return hashmap;
+}
+
+/* 销毁哈希表 */
+void freeArrayHashMap(ArrayHashMap* hashmap) {
+    for(int i = 0; i < hashmap->capacity; i++) {
```

Comment:
`for (`, 
Simply adjust the format.
It is recommended to use vscode's own formatting tool , configuration reference https://www.hello-algo.com/chapter_appendix/installation/#3-cc